cni.go

```diff
@@ -83,6 +89,11 @@ type libcni struct {
 	sync.RWMutex
 }
 
+type NetworkInterface struct {
```

Comment:
I am thinking through this type. It works in the current however give me some time to debate. If anyone likes the current state lmk